test/client_test.js

```diff
@@ -110,20 +110,34 @@ suite('api (client)', function() {
     ));
 
     let gotSuffixes = (result) =>
-      _.map(_.filter(result,
+      _.map(_.filter(result.clients,
         c => c.clientId.startsWith(CLIENT_ID)),
       c => c.clientId.substr(CLIENT_ID.length)).sort();
 
     // get all clients
-    assume(gotSuffixes(await helper.auth.listClients())).to.deeply.equal(suffixes);
+    let clients = await helper.auth.listClients();
+    assume(gotSuffixes(clients)).to.deeply.equal(suffixes);
 
     // prefix filtering
     assume(gotSuffixes(await helper.auth.listClients({prefix: CLIENT_ID + '/bb'})))
-      .to.deeply.equal(['/bb', '/bb/1', '/bb/2']);
+      .to.deeply.equal(['/bb', '/bb/1', '/bb/2', '/bb/3', '/bb/4', '/bb/5']);
     assume(gotSuffixes(await helper.auth.listClients({prefix: CLIENT_ID + '/bb/'})))
       .to.deeply.equal(['/bb/1', '/bb/2']);
     assume(gotSuffixes(await helper.auth.listClients({prefix: CLIENT_ID + '/c'})))
       .to.deeply.equal([]);
+    assume(gotSuffixes(await helper.auth.listClients({
+      prefix: CLIENT_ID + '/bb',
+      limit: 1,
+    }))).to.deeply.equal(['/bb']);
```

Comment:
Azure doesn't actually guarantee the order that results are returned.  So this could be /bb, or it could be /bb/3.